### PR TITLE
Add effect to TR5 Lasers

### PR DESCRIPTION
**Pull request by richardba**

### CODE REVIEW

**Comment by Nemoel-Tomo** [comment]
please give me time to review and check the shaders and effects

File: TombEngine/Shaders/Math.hlsli [hunk 1]
```diff
@@ -121,6 +125,42 @@ float SimplexNoise(float3 v)
...
+
+	float2 u = f * f * (3.0 - 2.0 * f);
+
+	return lerp(a, b, u.x) +
+							(c - a) * u.y * (1.0 - u.x) +
+							(d - b) * u.x * u.y;
+}
+
+#define OCTAVES 6
+float fractalNoise (float2 input)
```
Comment (Lwmte) [inline TombEngine/Shaders/Math.hlsli:147]: Please maintain PascalCase naming convention.

File: TombEngine/Shaders/Math.hlsli [hunk 2]
```diff
@@ -121,6 +125,42 @@ float SimplexNoise(float3 v)
...
+{
+	// Initial values
+	float value = 0.0;
+	float amplitude = .5;
+	float frequency = 0.;
+	//
+	// Loop of octaves
+	for (int i = 0; i < OCTAVES; i++)
+		{
```
Comment (Lwmte) [inline TombEngine/Shaders/Math.hlsli:156]: Please format properly, according to coding conventions.

File: TombEngine/Renderer/Renderer11.h [hunk 3]
```diff
@@ -601,24 +602,24 @@ namespace TEN::Renderer
...
+					 			 const Vector4& color0, const Vector4& color1, const Vector4& color2, const Vector4& color3, float orient2D,
+					 			 float scale, Vector2 size, BLEND_MODES blendMode, bool isSoftParticle, RenderView& view);
 		void AddColoredQuad(const Vector3& vertex0, const Vector3& vertex1, const Vector3& vertex2, const Vector3& vertex3,
-							const Vector4& color, BLEND_MODES blendMode, RenderView& view);
+												const Vector4& color, BLEND_MODES blendMode, RenderView& view);
 		void AddColoredQuad(const Vector3& vertex0, const Vector3& vertex1, const Vector3& vertex2, const Vector3& vertex3,
-							const Vector4& color0, const Vector4& color1, const Vector4& color2, const Vector4& color3,
-							BLEND_MODES blendMode, RenderView& view, bool isTr5Laser = false);
+												const Vector4& color0, const Vector4& color1, const Vector4& color2, const Vector4& color3,
+												BLEND_MODES blendMode, RenderView& view, bool isTr5Laser=false);
```
Comment (Lwmte) [inline TombEngine/Renderer/Renderer11.h:622]: Why `isTr5Laser=true` was condensed?

Comment (richardba) [reply TombEngine/Renderer/Renderer11.h]: Thanks, I fixed that!

File: TombEngine/Shaders/Math.hlsli (hunk 2 above)
Comment (richardba) [reply TombEngine/Shaders/Math.hlsli]: Solrry

File: TombEngine/Shaders/Math.hlsli (hunk 1 above)
Comment (richardba) [reply TombEngine/Shaders/Math.hlsli]: Thanks for the review!

**Comment by Kubsy** [comment]
Tomo took the commits and opened new pr so closing.

**Pull request closed without merging**